antsibull/docs_parsing/fqcn.py

```diff
@@ -23,6 +23,8 @@
 NAMESPACE_RE_STR = '[a-z0-9][a-z0-9_]+'
 #: Format of a FQCN
 FQCN_RE = re.compile(fr'({NAMESPACE_RE_STR})\.({NAMESPACE_RE_STR})\.(.*)')
+FQCN_STRICT_RE = re.compile(
+    fr'({NAMESPACE_RE_STR})\.({NAMESPACE_RE_STR})\.({NAMESPACE_RE_STR}(?:\.{NAMESPACE_RE_STR})*)')
```

Comment:
Maybe we should add examples to explain what the regex is designed to accept.

Reply:
I wrote a bit about the differences between the two regexes, and the regex used by Ansible itself in 688c7d03ebb9758a08922fa00df70748c4eab17c.